aten/src/ATen/native/cpu/ActivationKernel.cpp

```diff
@@ -0,0 +1,484 @@
+  }
+  return sum;
+}
+
+template <typename scalar_t>
+inline scalar_t
+_vec_sum_mul(int64_t dim_size, scalar_t* input_data, scalar_t* input_data2) {
+  using Vec = vec256::Vec256<scalar_t>;
+  int64_t d = 0;
+  Vec sum_vec(0);
+  scalar_t sum = 0;
+  for (; d < dim_size; d += Vec::size) {
+    if (d + Vec::size > dim_size) {
+      for (int i = d; i < dim_size; i++) {
+        sum += input_data[i] * input_data2[i];
+      }
+    } else {
+      Vec value(0);
+      Vec value2(0);
+      value.load(input_data + d);
+      value2.load(input_data2 + d);
+      value = value * value2;
+      sum_vec = sum_vec + value;
+    }
+  }
+  scalar_t sum_arr[Vec::size];
+  sum_vec.store(sum_arr);
+  for (int64_t i = 0; i < Vec::size; i++) {
+    sum += sum_arr[i];
+  }
+  return sum;
+}
+
+template <typename scalar_t>
+inline scalar_t _vec_max(int64_t dim_size, scalar_t* input_data) {
+  using Vec = vec256::Vec256<scalar_t>;
+  Vec max_val_vec(std::numeric_limits<scalar_t>::lowest());
+  scalar_t max_val = std::numeric_limits<scalar_t>::lowest();
+  for (int64_t d = 0; d < dim_size; d += Vec::size) {
+    int64_t leftover = _leftover<Vec::size>(d, dim_size);
+    Vec value;
+    value.load(input_data + d, leftover);
+    if (leftover < Vec::size) {
+      for (int i = 0; i < leftover; i++) {
+        if (input_data[d + i] > max_val) {
+          max_val = input_data[d + i];
+        }
+      }
+    } else {
+      max_val_vec = vec256::max(value, max_val_vec);
+    }
+  }
+  scalar_t max_val_arr[Vec::size];
+  max_val_vec.store(max_val_arr);
+  for (int64_t i = 0; i < Vec::size; i++) {
+    if (max_val_arr[i] > max_val) {
+      max_val = max_val_arr[i];
+    }
+  }
+  return max_val;
+}
+
+template <typename scalar_t>
+inline void _vec_mul_scalarsub_write(
+    scalar_t* grad_input_data,
+    scalar_t* grad_data,
+    scalar_t* output_data,
+    int64_t dim_size,
+    scalar_t sum) {
+  using Vec = vec256::Vec256<scalar_t>;
+  Vec sum_vec(sum);
+  int64_t d = 0;
+  for (; d < dim_size; d += Vec::size) {
+    Vec output(0);
+    Vec grad(0);
+    int64_t leftover = _leftover<Vec::size>(d, dim_size);
+    output.load(output_data + d, leftover);
+    grad.load(grad_data + d, leftover);
+    grad = grad - sum_vec;
+    grad = grad * output;
+
+    grad.store(grad_input_data + d, leftover);
+  }
+}
+
+template <typename scalar_t>
+inline void _vec_sub_exp_scalarmul_write(
+    scalar_t* grad_input_data,
+    scalar_t* grad_data,
+    scalar_t* output_data,
+    int64_t dim_size,
+    scalar_t sum) {
+  using Vec = vec256::Vec256<scalar_t>;
+  Vec sum_vec(sum);
+  int64_t d = 0;
+  for (; d < dim_size; d += Vec::size) {
+    Vec output(0);
+    Vec grad(0);
+    int64_t leftover = _leftover<Vec::size>(d, dim_size);
+    output.load(output_data + d, leftover);
+    grad.load(grad_data + d, leftover);
+    output = output.exp();
+    output = output * sum_vec;
+    grad = grad - output;
+
+    grad.store(grad_input_data + d, leftover);
+  }
+}
```

Comment:
Same issue here

Reply:
It should support any dim_size because load is passed leftover, which handles dim_sizes that are not a multiple of Vec::size.